gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,286 @@
+            err: err
+        });
+    }
+    return promiseTogo;
+};
+
+/**
+ * Fetches all settings for a particular solution, going through all of the solutions settings handlers.
+ *
+ * @param {lifecycleManager.invokeSettingsHandlerGet} invokeSettingsHandlerGet - Invoker from the `lifecycleManager`
+ * necessary for extracting settings from a solution and it's spec.
+ * @param {Object} solution - The solution registry entry body for this solution.
+ * @param {String} solutionID - The dotted solution id, example `com.microsoft.windows.cursors`.
+ * @param {lifecycleManager.variableResolver} resolver - Resolver for looking up handler specifications
+ * @return {Array} An array of promises resolving to the results from each settings handler the solution uses.
+ */
+gpii.flowManager.capture.fetchAllSettingsForSolution = function (invokeSettingsHandlerGet, solution, solutionID, resolver) {
+    var settingsPromisesTogo = [];
+
+    var newSolution = gpii.lifecycleManager.transformSolutionSettings(solution, gpii.settingsHandlers.changesToSettings);
+
+    // TODO GPII-4146  In progress refactoring back to gpii-windows
+    // https://issues.gpii.net/browse/GPII-4146
+    // if (solutionID === "com.freedomscientific.jaws") {
+    //     newSolution = gpii.flowManager.jawsVoiceProfileAdjustment(newSolution, localResolver);
+    // }
+
+    fluid.each(newSolution.settingsHandlers, function (handlerSpec) {
+        //if applicationSettings is unspecified, dont attempt to get settings
+        if (!handlerSpec.supportedSettings) {
+            return;
+        }
+        handlerSpec.settings = handlerSpec.supportedSettings;
+        // TODO There is some in progress refactoring that happened here to update spiSettingsHandler.
+        // No longer here, but this is a reminder to resolve it when there are acceptance tests for the SPI settings capture.
+        handlerSpec = resolver.resolve(handlerSpec, {});
+        settingsPromisesTogo.push(gpii.flowManager.capture.safeHandlerGet(invokeSettingsHandlerGet, solutionID, handlerSpec));
+    });
+    return settingsPromisesTogo;
+};
+
+/**
+ * Fetches all the settings for an entire list of solutions.
+ *
+ * @param {Object} solutionsToFetch - A keyed object of solutions using the same format as the solutions registry. Keys
+ * are solutionsIDs, like `com.freedomscientific.jaws` and the entries are the full solution registry entries.
+ * @param {lifecycleManager.invokeSettingsHandlerGet} invokeSettingsHandlerGet - Invoker from the `lifecycleManager`
+ * necessary for extracting settings from a solution and it's spec.
+ * @param {lifecycleManager.variableResolver} resolver - Resolver for looking up handler specifications
+ * @return {Array} Returns a list of promises that resolve to captured settings for each particular settings handler in
+ * every solution.
+ */
+gpii.flowManager.capture.fetchAllSettingsForSolutionsList = function (solutionsToFetch, invokeSettingsHandlerGet, resolver) {
```

Comment:
With the recent merge of the new read API, this function and [`lifecycleManager.read()`}(https://github.com/GPII/universal/blob/master/gpii/node_modules/lifecycleManager/src/LifecycleManager.js#L1336) are pretty much the same. Is it possible to directly use `lifecycleManager.read()` or at least consolidate the common part into a shared function?

Reply:
Hi @cindyli , as you mentioned the code between these two features are identical so I was able to remove mine and just use the lifecyclemanager. However, as soon as GPII-3119 is merged in (hopefully today or tomorrow), we can remove the custom SPI settings handling from the lifecycle manager processRead method.

Reply:
Agree. Thanks for the heads up.